### PR TITLE
Remove edit/upgrade action from Apps with no chart

### PR DESCRIPTION
**Pull request by jordojordo**
This is in reference to rancher/dashboard#4120 where an App created with the old UI would send an error when trying to edit/update: `Cannot read property 'annotations' of undefined`.

According to https://rancher.com/docs/rancher/v2.6/en/helm-charts/ , any App that has been managed by the Cluster Manager (old UI) should continue to be managed only by the Cluster Manager. 

The actual error in this case with OpenEBS was caused by the App utilizing a chart (`2.5.1`) that no longer exists in the default repositories. This PR will check if an App has a chart that is not within the repositories and will disable the `Edit/Upgrade` option from the action menu.

### ** Update: https://github.com/rancher/dashboard/pull/4332#issuecomment-941436439

To reproduce:
1. Navigate to `localhost/g/apps` and launch OpenEBS.
 
oldAppLaunch

2. Navigate to Cluster > Apps & Marketplace > Installed Apps

installedApps

3. Then select the OpenEBS App and try to edit. 

oldAppNoEditAction

### CODE REVIEW

File: assets/translations/en-us.yaml [hunk 1]
```diff
@@ -718,6 +718,14 @@ catalog:
             other { seconds }
           }
       wait: Wait
+    multiCluster:
+      label: This is a Multi-cluster App and it cannot be modified here.
```
Comment (nwmac) [inline assets/translations/en-us.yaml:722]: We shouldn't split localisations like this - not all languages necessarily split in the same was as English. I think you should be able to use the `<nuxt-link>` element in the string and pass the target route as an arg - much line we do with `<a href ` - e.g. see the string `v1ClusterTools.monitoring.description`

File: pages/c/_cluster/apps/charts/install.vue [hunk 2]
```diff
@@ -103,6 +104,12 @@ export default {
       this.forceNamespace = null;
     }
 
+    const isMultiClusterApp = await this.existing.deployedAsMultiCluster();
```
Comment (nwmac) [inline pages/c/_cluster/apps/charts/install.vue:107]: You should be able to just do `this.mcapp = await this.existing.deployedAsMultiCluster();`

**Review by nwmac — CHANGES_REQUESTED** [review]
@jordojordo Can we go through this one - I could not reproduce. I think we might also need to look at the legacy catalog as well as multi-cluster apps

**Comment by rmweir** [comment]
@jordojordo @vincent99 what was the conclusion here? I think we need to disable upgrade/edit for apps that were originally created through ember. They upgrade/edit will not work a majority of the time. The version that is automatically found has been wrong every time I have seen it. This is because a chart from a repo was installed, and there is no clusterrepo equivalent of that repo but there is one that contains some other different implementation of the chart with totally different versions. I think the only way to identify these is to check if it has the repo annotation or not and if not then those options should be disabled. @cbron @cloudnautique and I discussed about this and actually thought maybe we should gray out the entire app in this case.

**Comment by jordojordo** [comment]
@rmweir This is a work in progress. Currently I have disabled editing/upgrading of Apps that have been created through Ember that are considered Multi-Cluster Apps. What still needs to be done is filtering the Apps on a cluster/project level that have also been created with Ember.

**Comment by nwmac** [comment]
@jordojordo I've updated your PR:

- Install was broken since you were accessing properties of existing which was null
- Clicking on links caused the page to reload, rather than doing an in-app navigation
- Multi-cluster apps were not shows as such - from my testing, a mc app has both a legacy and mc app, so your logic was preferring legacy - I've changed that around

Can you review my changes and merge if you are okay with them

**Review by nwmac — APPROVED** [review]

**Comment by jordojordo** [comment]
@nwmac Thanks for making those changes, they make sense and LGTM. I updated the `deployedAsMultiCluster` method as well.

File: models/catalog.cattle.io.app.js [hunk 3]
```diff
@@ -215,13 +215,7 @@ export default class CatalogApp extends SteveModel {
...
 
       if (mcapps) {
-        for (let i = 0 ; i < mcapps.length ; i++) {
-          const res = mcapps[i].spec?.targets?.find(target => target.appName === this.metadata?.name);
-
-          if (res) {
-            return res;
-          }
-        }
+        return mcapps.find(mcapp => mcapp.spec.targets.find(target => target.appName === this.metadata?.name));
```
Comment (nwmac) [inline models/catalog.cattle.io.app.js:218]: We've lost the ? on spec.targets - I'd put that back in, just in case.

**Review by nwmac — CHANGES_REQUESTED** [review]

File: models/catalog.cattle.io.app.js (hunk 3 above)
Comment (jordojordo) [reply models/catalog.cattle.io.app.js]: I've reapplied the `spec?.targets?` that was there before.

**Review by nwmac — APPROVED** [review]
LGTM

**Pull request merged**